components/push/android/src/main/java/org/mozilla/push/PushConnection.kt

```diff
@@ -13,102 +13,84 @@ import java.io.File
 /**
  * An implementation of a [PushAPI] backed by a Rust Push library.
  *
- * @param path an absolute path to a file that will be used for the internal database.
+ * @param server_host the host name for the service (e.g. "push.service.mozilla.org").
+ * @param socket_protocol the optional socket protocol (default: "https")
+ * @param bridge_type the optional bridge protocol (default: "fcm")
+ * @param application_id the native OS messaging registration id
  * @param encryption_key an optional key used for encrypting/decrypting data stored in the internal
  *  database. If omitted, data will be stored in plaintext.
- //* /
-
-class PushConnection(path: String, encryption_key: String? = null) : PushAPI, AutoCloseable {
-    private var db: RawPushConnection?
+ */
+class PushConnection(
+    application_id: String,
+    sender_id: String,
+    server_host: String?="push.service.mozilla.org",
+    socket_protocol: String?="https",
+    bridge_type:String?="fcm",
+    encryption_key: String? = null) : PushAPI, AutoCloseable {
+    private var conn: RawPushConnection?
```

Comment:
From my limited understanding, I believe this is a long-lived socket connection. If so, is there an automatic retry?

Reply:
It's not quite a long lived connection, at least not on mobile.

Mobile will be using whatever the native messaging service offers. There are some calls that need to be made to the server, and those will be handled by the push connection. We're also going to wrap these inside of a "PushManager" which will handle more things like decryption and managing the individual connections.

As far as you should be concerned, there should be a fairly small set of functions that you'll need to call. You will have to keep some state, though, for things like which ChannelID goes to what callback, but ideally, we'll try to keep as much of that out of your hair as possible.